scripts/tests/run_python_test.py

```diff
@@ -80,36 +80,52 @@ def DumpProgramOutputToQueue(thread_list: typing.List[threading.Thread], tag: st
                                            (f"[{tag}][\33[31mSTDERR\33[0m]").encode(), queue))
 
 
-@click.command()
-@click.option("--app", type=str, default=None, help='Local application to use, omit to use external apps, use a path for a specific binary or use a filename to search under the current matter checkout.')
-@click.option("--factoryreset", is_flag=True, help='Remove app config and repl configs (/tmp/chip* and /tmp/repl*) before running the tests.')
-@click.option("--app-params", type=str, default='', help='The extra parameters passed to the device.')
-@click.option("--script", type=click.Path(exists=True), default=FindBinaryPath("mobile-device-test.py"), help='Test script to use.')
```

Comment:
I think this is what seems slow: finding a binary path. We can remove it or move it somewhere else to not have a default and then it would be just as fast and more readable than argparse.

Reply:
Yes, it is verified that FindBinaryPath is very slow on darwin (it returns almost immediately on Linux)